core/src/mindustry/type/UnitType.java

```diff
@@ -1339,25 +1344,29 @@ public Color shieldColor(Unit unit){
         return shieldColor == null ? unit.team.color : shieldColor;
     }
 
-
     public void drawMining(Unit unit){
+        if(drawMineBeam){
+            float focusLen = mineBeamOffset + Mathf.absin(Time.time, 1.1f, 0.5f);
+            float px = unit.x + Angles.trnsx(unit.rotation, focusLen);
+            float py = unit.y + Angles.trnsy(unit.rotation, focusLen);
+
+            drawMiningBeam(unit, px, py);
+        }
+    }
+
+    public void drawMiningBeam(Unit unit, float px, float py){
         if(!unit.mining()) return;
-        float focusLen = unit.hitSize / 2f + Mathf.absin(Time.time, 1.1f, 0.5f);
         float swingScl = 12f, swingMag = tilesize / 8f;
         float flashScl = 0.3f;
 
-        float px = unit.x + Angles.trnsx(unit.rotation, focusLen);
-        float py = unit.y + Angles.trnsy(unit.rotation, focusLen);
-
         float ex = unit.mineTile.worldx() + Mathf.sin(Time.time + 48, swingScl, swingMag);
         float ey = unit.mineTile.worldy() + Mathf.sin(Time.time + 48, swingScl + 2f, swingMag);
 
         Draw.z(Layer.flyingUnit + 0.1f);
 
         Draw.color(Color.lightGray, Color.white, 1f - flashScl + Mathf.absin(Time.time, 0.5f, flashScl));
 
-        Draw.alpha(Renderer.unitLaserOpacity);
-        Drawf.laser(mineLaserRegion, mineLaserEndRegion, px, py, ex, ey, 0.75f);
+        Drawf.laser(Core.atlas.find("minelaser"), Core.atlas.find("minelaser-end"), px, py, ex, ey, 0.75f);
```

Comment:
Is this accidental? You've removed the opacity code that you added a setting for, and the proper region usage too.

Reply:
Oops

Reply:
Copied code from the old PR